FluentTc.Tests/ProjectRetrievalTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Collections.Generic;
+using FakeItEasy;
+using FluentTc.Domain;
+using FluentTc.Engine;
+using NUnit.Framework;
+
+namespace FluentTc.Tests
+{
+    [TestFixture]
+    public class ProjectRetrievalTests
+    {
+        [TestCase("FluentTC")]
+        [TestCase("Test 1")]
+        [TestCase("Test%20C")]
+        public void GetProject_ByName(string projectName)
+        {
+            // Arrange
+            var teamCityCaller = CreateTeamCityCaller();
+            new RemoteTc().Connect(_ => _.AsGuest(), teamCityCaller)
+                          .GetProject(project => project.Name(projectName));
+
+            //Expectations
+            var expectedCall = string.Format(@"/app/rest/projects/name:{0}", projectName);
+            A.CallTo(() => teamCityCaller.Get<Project>(expectedCall)).MustHaveHappened();
+        }
+
+        [TestCase("1234")]
+        [TestCase("ABcd")]
+        [TestCase("129@$$jd")]
+        public void GetProject_ById(string projectId)
+        {
+            // Arrange
+            var teamCityCaller = CreateTeamCityCaller();
+            new RemoteTc().Connect(_ => _.AsGuest(), teamCityCaller)
+                          .GetProject(project => project.Id(projectId));
+
+            //Expectations
+            var expectedCall = string.Format(@"/app/rest/projects/id:{0}", projectId);
+            A.CallTo(() => teamCityCaller.Get<Project>(expectedCall)).MustHaveHappened();
+        }
+
+        private static TeamCityCaller CreateTeamCityCaller()
```

Comment:
I started with that and realized I was not mocking it the right way. So moved that out and new tests should be following the mocking correctly.